src/report/azure-devops.mjs

```diff
@@ -0,0 +1,103 @@
+/*
+Message grouping
+##[group]Beginning of a group
+##[warning]Warning message
+##[error]Error message
+##[section]Start of a section
+##[debug]Debug text
+##[command]Command-line being run
+##[endgroup]
+
+Warnings and errors:
+
+##vso[task.logissue type=warning;sourcepath=consoleapp/main.cs;linenumber=1;columnnumber=1;code=100;]Found something that could be a problem.
+
+Progress
+##vso[task.setprogress]current operation
+
+Complete
+##vso[task.complete]current operation
+*/
+
+import { EOL } from "node:os";
+
+const SEVERITY2VSO_TYPE = new Map([
+  // "error" | "warn" | "info" | "ignore
+  ["error", "error"],
+  ["warn", "warning"],
+  // azure devops doesn't seem to understand 'info'. We still want to
+  // show them, though, hence:
+  ["info", "warning"],
+]);
+
+/**
+ *
+ * @param {import("../../types/shared-types.js").SeverityType} pSeverity
+ */
+function formatSeverity(pSeverity) {
+  return SEVERITY2VSO_TYPE.get(pSeverity) ?? "warning";
+}
+
+/**
+ *
+ * @param {import("../../types/violations.js").IViolation} pViolation
+ */
+function formatViolation(pViolation) {
+  return `##vso[task.logissue type=${formatSeverity(
+    pViolation.rule.severity
+  )};sourcepath=${pViolation.from}]${pViolation.rule.name}`;
```

Comment:
I wonder if it would be useful to add `pViolation.to` to the message alongside `pViolation.rule.name`.
It could help understanding the details when there are multiple violations of the same rule originating from one file.

Reply:
Yes it is - it's on the todolist.

Reply:
...also for circular dependencies: the whole cycle, for 'via' rules the path found etc - it should be on-par with the `err` reporter.

Reply:
- [x] added those as well (see added commits) => will be part of the next beta